src/config/config.h

```diff
@@ -0,0 +1,198 @@
+    CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_FILMSTRIP_OVERLAY,
+    CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_WORKAROUND_BUGS,
+    CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_IMAGE_QUALITY,
+    CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_CACHE_DIR_ENABLED,
+    CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_CACHE_DIR,
+#endif
+    CFG_SERVER_EXTOPTS_MARK_PLAYED_ITEMS_ENABLED,
+    CFG_SERVER_EXTOPTS_MARK_PLAYED_ITEMS_STRING_MODE_PREPEND,
+    CFG_SERVER_EXTOPTS_MARK_PLAYED_ITEMS_STRING,
+    CFG_SERVER_EXTOPTS_MARK_PLAYED_ITEMS_SUPPRESS_CDS_UPDATES,
+    CFG_SERVER_EXTOPTS_MARK_PLAYED_ITEMS_CONTENT_LIST,
+#ifdef HAVE_LASTFMLIB
+    CFG_SERVER_EXTOPTS_LASTFM_ENABLED,
+    CFG_SERVER_EXTOPTS_LASTFM_USERNAME,
+    CFG_SERVER_EXTOPTS_LASTFM_PASSWORD,
+#endif
+    CFG_IMPORT_HIDDEN_FILES,
+    CFG_IMPORT_FILESYSTEM_CHARSET,
+    CFG_IMPORT_METADATA_CHARSET,
+    CFG_IMPORT_PLAYLIST_CHARSET,
+#ifdef HAVE_JS
+    CFG_IMPORT_SCRIPTING_CHARSET,
+    CFG_IMPORT_SCRIPTING_COMMON_SCRIPT,
+    CFG_IMPORT_SCRIPTING_PLAYLIST_SCRIPT,
+    CFG_IMPORT_SCRIPTING_PLAYLIST_SCRIPT_LINK_OBJECTS,
+    CFG_IMPORT_SCRIPTING_IMPORT_SCRIPT,
+#endif // JS
+    CFG_IMPORT_SCRIPTING_VIRTUAL_LAYOUT_TYPE,
+#ifdef HAVE_MAGIC
+    CFG_IMPORT_MAGIC_FILE,
+#endif
+    CFG_IMPORT_AUTOSCAN_TIMED_LIST,
+#ifdef HAVE_INOTIFY
+    CFG_IMPORT_AUTOSCAN_USE_INOTIFY,
+    CFG_IMPORT_AUTOSCAN_INOTIFY_LIST,
+#endif
+    CFG_IMPORT_MAPPINGS_IGNORE_UNKNOWN_EXTENSIONS,
+    CFG_IMPORT_MAPPINGS_EXTENSION_TO_MIMETYPE_CASE_SENSITIVE,
+    CFG_IMPORT_MAPPINGS_EXTENSION_TO_MIMETYPE_LIST,
+    CFG_IMPORT_MAPPINGS_MIMETYPE_TO_UPNP_CLASS_LIST,
+    CFG_IMPORT_MAPPINGS_MIMETYPE_TO_CONTENTTYPE_LIST,
+#ifdef HAVE_LIBEXIF
+    CFG_IMPORT_LIBOPTS_EXIF_AUXDATA_TAGS_LIST,
+#endif
+#ifdef HAVE_EXIV2
+    CFG_IMPORT_LIBOPTS_EXIV2_AUXDATA_TAGS_LIST,
+#endif
+#if defined(HAVE_TAGLIB)
+    CFG_IMPORT_LIBOPTS_ID3_AUXDATA_TAGS_LIST,
+#endif
+    CFG_TRANSCODING_PROFILE_LIST,
+#ifdef HAVE_CURL
+    CFG_EXTERNAL_TRANSCODING_CURL_BUFFER_SIZE,
+    CFG_EXTERNAL_TRANSCODING_CURL_FILL_SIZE,
+#endif //HAVE_CURL
+#ifdef SOPCAST
+    CFG_ONLINE_CONTENT_SOPCAST_ENABLED,
+    CFG_ONLINE_CONTENT_SOPCAST_REFRESH,
+    CFG_ONLINE_CONTENT_SOPCAST_UPDATE_AT_START,
+    CFG_ONLINE_CONTENT_SOPCAST_PURGE_AFTER,
+#endif
+#ifdef ATRAILERS
+    CFG_ONLINE_CONTENT_ATRAILERS_ENABLED,
+    CFG_ONLINE_CONTENT_ATRAILERS_REFRESH,
+    CFG_ONLINE_CONTENT_ATRAILERS_UPDATE_AT_START,
+    CFG_ONLINE_CONTENT_ATRAILERS_PURGE_AFTER,
+    CFG_ONLINE_CONTENT_ATRAILERS_RESOLUTION,
+#endif
+#if defined(HAVE_FFMPEG)
+    CFG_IMPORT_LIBOPTS_FFMPEG_AUXDATA_TAGS_LIST,
+#endif
+    CFG_MAX
+} config_option_t;
+
+class Config {
+public:
+    /// \brief Returns the name of the config file that was used to launch the server.
+    virtual fs::path getConfigFilename() const = 0;
+
+    /// \brief returns a config option of type std::string
+    /// \param option option to retrieve.
+    virtual std::string getOption(config_option_t option) = 0;
```

Comment:
Do you want to take an opportunity and make them `const`? So that all other methods can take `shared_ptr<const Config>`? No one really changes them, right?